interpreter/go/go.go

```diff
@@ -11,16 +11,17 @@ package golang // import "go.opentelemetry.io/ebpf-profiler/interpreter/go"
 import "C"
 
 import (
+	"debug/gosym"
```

Comment:
I believe `debug/gosym` will allocate O(n) memory for the symbols regardless of its mmapped or not. We may need to reimplement its functionality in `elfgopclntab` in a way that does not use additional memory.

Have you looked at memory usage characteristics of this with large go executables?

Reply:
This draft just serves to showcase potential benefits of https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/455. I didn't look further into memory opitmizations and also not, on why the coredump tests on arm64 are not symbolizing Go frames.
But the things you have pointed out are next steps once https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/455 is accpeted.